pyoptsparse/pyOpt_history.py

```diff
@@ -630,29 +630,32 @@ def getValues(self, names=None, callCounters=None, major=True, scale=False, stac
             callCounters.append(self.read("last"))
             callCounters.remove("last")
 
+        self._ipoptIterCounter = -1  # track iteration, only relevant for IPOPT
```

Comment:
Current method should work for other optimizers as well (will double check), I just put IPOPT in the variable/method names so that what I intend here is clear. But I will rename those.
I don't know if we want to have another pre-processing loop, as we are already doing some `callCounter` validation in the main loop (func vs funcSense, major, fail)

Reply:
Yeah I was thinking of creating a loop before we parse the file, where we generate the list of callCounters given all the various input flags. Then we only loop over those. But maybe that would decrease the efficiency so maybe it's not worth it. I'm okay with keeping it as is, but maybe some minor refactor would be helpful:
- lump all the checks into a single function that validates the callCounter against things like `funcs`, major flags etc
- make it general for all optimizers

Reply:
Refactored